modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -527,6 +527,37 @@ public ObjectDefinition enableAccountEntryRestricted(
 		return objectDefinitionPersistence.update(objectDefinition2);
 	}
 
+	@Override
+	public ObjectDefinition enableAccountEntryRestrictedForExternalStorageType(
+			ObjectField objectField)
+		throws PortalException {
+
+		if (!Objects.equals(objectField.getBusinessType(), "Integer") &&
+			!Objects.equals(objectField.getBusinessType(), "LongInteger") &&
+			!Objects.equals(objectField.getBusinessType(), "Text")) {
+
+			throw new ObjectDefinitionAccountEntryRestrictedException(
+				"Custom object definitions can only be restricted by a " +
+					"Integer, Long Integer or Text field");
+		}
+
+		ObjectDefinition objectDefinition = getObjectDefinition(
+			objectField.getObjectDefinitionId());
+
+		if (objectDefinition.isDefaultStorageType() ||
+			objectDefinition.isAccountEntryRestricted()) {
```

Comment:
How to change the accountEntryRestrictedObjectFieldId for an objectDefinition preconfigured as accountEntryRestricted?